packages/grafana-prometheus/src/query_hints.ts

```diff
@@ -230,6 +210,125 @@ export function getInitHints(datasource: PrometheusDatasource): QueryHint[] {
   return hints;
 }
 
+export function getExpandRulesHints(query: string, mapping: RuleQueryMapping): QueryHint[] {
+  const hints: QueryHint[] = [];
+  const mappingForQuery = Object.keys(mapping).reduce((acc, ruleName) => {
+    if (query.search(ruleName) === -1) {
+      return acc;
+    }
+
+    if (mapping[ruleName].length > 1) {
+      const { idx, expandedQuery, identifier, identifierValue } = getRecordingRuleIdentifierIdx(
+        query,
+        ruleName,
+        mapping[ruleName]
+      );
+
+      // No identifier detected add warning
+      if (idx === -1) {
+        hints.push({
+          type: 'EXPAND_RULES_WARNING',
+          label: 'Query contains a recording rule. To be able to expand it please add an identifier label/value.',
+        });
+        return acc;
+      } else {
+        // Identifier found.
+        return {
+          ...acc,
+          [ruleName]: {
+            expandedQuery,
+            identifier,
+            identifierValue,
+          },
+        };
+      }
+    } else {
+      return {
+        ...acc,
+        [ruleName]: {
+          expandedQuery: mapping[ruleName][0].query,
+        },
+      };
+    }
+  }, {});
+
+  if (size(mappingForQuery) > 0) {
+    const label = 'Query contains recording rules.';
+    hints.push({
+      type: 'EXPAND_RULES',
+      label,
+      fix: {
+        label: 'Expand rules',
+        action: {
+          type: 'EXPAND_RULES',
+          query,
+          options: mappingForQuery,
+        },
+      },
+    });
+  }
+
+  return hints;
+}
+
+export function getRecordingRuleIdentifierIdx(
+  queryStr: string,
+  ruleName: string,
+  mapping: RuleQueryMapping[string]
+): RecordingRuleIdentifier & { idx: number } {
+  const { query } = buildVisualQueryFromString(queryStr);
+  const queryMetricLabels: QueryBuilderLabelFilter[] = getQueryLabelsForRuleName(ruleName, query);
+  if (queryMetricLabels.length === 0) {
+    return { idx: -1, identifier: '', identifierValue: '', expandedQuery: '' };
+  }
+
+  let uuidLabel = '';
+  let uuidLabelValue = '';
+  let uuidLabelIdx = -1;
+
+  queryMetricLabels.forEach((qml) => {
+    if (uuidLabelIdx === -1 && qml.label.search('uuid') !== -1) {
```

Comment:
question: is it worth trying to generalize this and work with other labels that don't include "uuid" in the label name?

For Grafana SLO use cases, the "uuid" is sufficient. Maybe this is an easier place to start and we can generalize it later if we care to ?

Reply:
Not in this PR but maybe in a separate task we should definitely take a look at that. Do you have a suggestion how we would detect the identifier label?